cmd/auth-provider-gcp/plugin/plugin_test.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package plugin
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp/gcpcredential"
+	credentialproviderapi "k8s.io/kubelet/pkg/apis/credentialprovider"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+)
+
+const (
+	serviceAccountsEndpoint = "/computeMetadata/v1/instance/service-accounts/"
+	defaultEndpoint         = "/computeMetadata/v1/instance/service-accounts/default/"
+	scopeEndpoint           = defaultEndpoint + "scopes"
+	emailEndpoint           = defaultEndpoint + "email"
+	tokenEndpoint           = defaultEndpoint + "token"
+	dummyToken              = "ya26.lots-of-indiscernible-garbage"
+	email                   = "1234@project.gserviceaccount.com"
+	expectedUsername        = "_token"
+)
+
+func hasURL(url string, response *credentialproviderapi.CredentialProviderResponse) bool {
+	if _, ok := response.Auth[url]; ok {
+		return true
+	}
+	return false
+}
+
+func usernameMatches(expectedUsername string, auth credentialproviderapi.AuthConfig) bool {
+	return auth.Username == expectedUsername
+}
+
+func passwordMatches(expectedPassword string, auth credentialproviderapi.AuthConfig) bool {
+	return auth.Password == expectedPassword
+}
```

Comment:
nit: Not sure these functions make the code more readable than just inlining the logic.

Reply:
Ah, I was wondering if that might be the case - I'll inline them.